### DISCUSSION TITLE
Ability to specify number of columns

### DISCUSSION THREAD

**Discussion by Mosnar** [solspace/craft-express-forms#60] [Feature Requests]
### Description
Everything about this plugin is amazing except for one massive deal breaker for us: the ability to denote how many columns a field consumes. This does not need to be as robust as Freeform, but I can think of no time this plugin is usable for us without being able to put certain fields on the same row. Is this on the roadmap?

**Comment by kjmartens** [comment]
@Mosnar I assume you wish to take advantage of the automated rendering of forms then (since you can of course manually code forms to be however you want)? And if so, it's no limitation to you that you can't set up multi-option fields, etc (as those of course can't be automated then)?

That aside, how do you envision this working? (what does it look like to configure inside the CP, etc) 🙂

**Comment by Mosnar** [comment]
Exactly, the idea of purchasing a form plugin for our projects is that the client or our marketing team should be able to immediately create a form themselves without waiting on a developer to build out the template. You could enhance the drag-and-drop editor to add a handle to resize the width of a particular field and then drop a field in next to it; or, to keep things simpler, you could just flex the fields to fit the row they're dropped in. Fortunately, the design doesn't have to change too much. I quickly mocked that up in the Chrome inspector:
![image](https://user-images.githubusercontent.com/3810939/57087860-00dca100-6ccf-11e9-8ac1-1b572393ad3a.png)
![image](https://user-images.githubusercontent.com/3810939/57087986-41d4b580-6ccf-11e9-85c6-6bde821d0672.png)

**Comment by kjmartens** [comment]
Thanks for your feedback @Mosnar. That's the exact solution we provide with Freeform. The purpose of Express Forms was to offer a solution that was more affordable for simpler or smaller sites/needs and/or developers that enjoy the full control of manually building out their forms (or heavily extending the plugin's features).

Is there something specific about Freeform that is holding you back from using that instead? 🙂

**Comment by Mosnar** [comment]
I understand, it's just a bit expensive for clients with smaller contracts. Maybe as a middle ground, form fields could be given some arbitrary metadata (say, for the number of columns a field consumes) so a template could theoretically be built to accommodate those designs.

**Comment by kjmartens** [comment]
Understandable. We do have some near future plans for Freeform that will help make Freeform a little more affordable (can't say anything more on that at the moment). In the meantime, we will definitely consider your request to have options to store metadata for fields. 🙂
